### PR TITLE
Seen phrases

### PR DESCRIPTION
**Pull request by lightme-fan**
Hello team,

This is a feature that lets the users know all phrases they have seen and allows them learn those phrases. All the changes are in this [branch link](https://github.com/onja-org/malagasy-rn-Bismuth/tree/seen-phrases). Please, have a look at it.
This is the [click up](https://app.clickup.com/t/35bjb7) link.

Thank you in advance

### CODE REVIEW

File: App.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react';
 import Navigation from './src/Navigation/index';
 import {Provider} from 'react-redux';
-import configureStore from './src/redux/store';
+import configureStore, {seenPhraseStore} from './src/redux/store';
```
Comment (lightme-fan) [inline App.js]: Remove unused code

File: src/screens/Learning.js [hunk 2]
```diff
@@ -47,11 +53,12 @@ export default ({
   };
 
   const selectAnswerCallback = useCallback(
-    item => {
+    async item => {
```
Comment (lightme-fan) [inline src/screens/Learning.js]: Remove this async keyword. It is not used here.

File: src/screens/Learning.js [hunk 3]
```diff
@@ -34,7 +40,7 @@ export default ({
   const [disableAllOptions, setDisableAllOptions] = useState(false);
   const [shouldReshuffle, setshouldReshuffle] = useState(false);
 
-  useEffect(() => {
+  useEffect(async () => {
```
Comment (lightme-fan) [inline src/screens/Learning.js]: Remove this async keyword. It is not used here.

File: src/screens/Learning.js [hunk 4]
```diff
@@ -19,13 +20,18 @@ import ModeIcon from '../components/ToolButton/assets/mode.svg';
...
 import {LANGUAGE_NAMES} from '../data/dataUtils';
 import {shuffleArray} from '../utils';
+import {currentCategory} from '../redux/selectors';
+import {seenPhrasesId} from './Home';
 
 export default ({
   //nav provider
   navigation,
-
+  categories,
```
Comment (lightme-fan) [inline src/screens/Learning.js:29]: This is unused, remove it from here.

**Review by lightme-fan — COMMENTED** [review]
You have a few unused code. Remove all of them. Thanks!

File: src/screens/Learning.js [hunk 5]
```diff
@@ -133,7 +145,11 @@ export default ({
           </View>
           <View style={styles.heading}>
             <SectionHeading text="Category: " />
-            <Text>{currentCategoryName}</Text>
+            <Text>
+              {currentCategoryId === seenPhrasesId
+                ? `Seen phrases - ${currentSeenPhraseCategory?.name.en}`
```
Comment (lightme-fan) [inline src/screens/Learning.js]: Change ${currentSeenPhraseCategory?.name.en} into ${currentSeenPhraseCategory?.name.EN}

**Review by lightme-fan — COMMENTED** [review]
You still used en instead of EN in the category name.

**Review by Tiaraha-funny — APPROVED** [review]

File: src/screens/Home.js [hunk 6]
```diff
@@ -49,6 +55,18 @@ export default ({
     navigation.navigate('Learn');
   };
 
+  const openSeenPhrases = async item => {
+    setCurrentCategory(item.id);
+    setPhrases(seenPhrases);
+    if (seenPhrases.length === 0) {
+      alert(
+        'No phrases you have seen yet. Please, learn new phrases again. Thanks',
+      );
```
Comment (andrzej-onja) [inline src/screens/Home.js]: remove alert just do no action

File: src/screens/Home.js [hunk 7]
```diff
@@ -113,12 +131,19 @@ export default ({
             <SectionHeading text="Seen phrases:" />
           </View>
           <List
-            data={[{id: 1, name: '35 words and phrases'}]}
+            data={[
+              {
+                id: '#seen-phrases',
```
Comment (andrzej-onja) [inline src/screens/Home.js]: use created const for the string

File: src/screens/Learning.js [hunk 8]
```diff
@@ -51,7 +56,8 @@ export default ({
       if (item.id === currentPhrase.id) {
         // TODO add to learned
       } else {
-        // TODO add to seen
+        seenPhrases.every(phrase => phrase.id !== currentPhrase.id) &&
+          addSeenPhrases(item);
```
Comment (andrzej-onja) [inline src/screens/Learning.js]: addSeenPhrase**s** ? its a single phrase, remove s

File: src/useAsyncStorage/index.js [hunk 9]
```diff
@@ -0,0 +1,21 @@
+import AsyncStorage from '@react-native-community/async-storage';
+export const SEEN_PHRASE_KEY = '@seenPhrases';
+
+// Setting data to storage
+export const setSeenPhrasesToStorage = async (key, phrases) => {
```
Comment (andrzej-onja) [inline src/useAsyncStorage/index.js]: these names should be universal 
`setSeenPhrasesToStorage
getSeenPhraseFromStorage`

like 


`
setItemToStorage
getItemFromStorage`

**Review by andrzej-onja — CHANGES_REQUESTED** [review]

**Review by ravinalamada — APPROVED** [review]

**Pull request closed without merging**